main/unbound/unbound.initd

```diff
@@ -22,36 +23,20 @@ depend() {
 	after auth-dns
 }
 
-checkconfig() {
-	checkpath --directory /var/run/unbound
-	UNBOUND_PIDFILE=$("${UNBOUND_CHECKCONF}" -o pidfile "${UNBOUND_CONFFILE}")
-	return $?
-}
-
 configtest() {
-	ebegin "Checking ${SVCNAME} configuration"
-	checkconfig
-	eend $?
-}
-
-start() {
-	checkconfig || return $?
-	ebegin "Starting ${SVCNAME}"
-	start-stop-daemon --start --pidfile "${UNBOUND_PIDFILE}" \
-		--exec "${UNBOUND_BINARY}" -- -c "${UNBOUND_CONFFILE}"
+	ebegin "Checking $RC_SVCNAME configuration"
+	"$checkcfg" "$cfgfile" 1>/dev/null || return $?
 	eend $?
 }
 
-stop() {
-	checkconfig || return $?
-	ebegin "Stopping ${SVCNAME}"
-	start-stop-daemon --stop --pidfile "${UNBOUND_PIDFILE}"
-	eend $?
+start_pre() {
+	checkpath --directory /run/unbound
+	configtest || return $?
```

Comment:
the `|| return $?` after `configtest` should be included - I tested the `reload` command with a config file with errors:

```
musl64 [~/aports/main/unbound]$ sudo rc-service unbound reload
 * Checking unbound configuration ...
/etc/unbound/unbound.conf:11: error: unknown keyword 'include'
/etc/unbound/unbound.conf:11: error: stray '"'
/etc/unbound/unbound.conf:11: error: unknown keyword '/etc/unbound/local.d/*.conf'
/etc/unbound/unbound.conf:11: error: stray '"'
read /etc/unbound/unbound.conf failed: 4 errors in configuration file
 * Reloading unbound ...
start-stop-daemon: option requires an argument: pidfile
Usage: start-stop-daemon [options] 

Options: [ I:KN:PR:Sa:bc:d:e:g:ik:mn:op:s:tu:r:w:x:1:2:ChqVv ]
  -I, --ionice <arg>                Set an ionice class:data when starting
  -K, --stop                        Stop daemon
  -N, --nicelevel <arg>             Set a nicelevel when starting
  -R, --retry <arg>                 Retry schedule to use when stopping
  -S, --start                       Start daemon
  -a, --startas <arg>               deprecated, use --exec or --name
  -b, --background                  Force daemon to background
  -c, --chuid <arg>                 deprecated, use --user
  -d, --chdir <arg>                 Change the PWD
  -e, --env <arg>                   Set an environment string
  -k, --umask <arg>                 Set the umask for the daemon
  -g, --group <arg>                 Change the process group
  -i, --interpreted                 Match process name by interpreter
  -m, --make-pidfile                Create a pidfile
  -n, --name <arg>                  Match process name
  -o, --oknodo                      deprecated
  -p, --pidfile <arg>               Match pid found in this file
  -s, --signal <arg>                Send a different signal
  -t, --test                        Test actions, don't do them
  -u, --user <arg>                  Change the process user
  -r, --chroot <arg>                Chroot to this directory
  -w, --wait <arg>                  Milliseconds to wait for daemon start
  -x, --exec <arg>                  Binary to start/stop
  -1, --stdout <arg>                Redirect stdout to file
  -2, --stderr <arg>                Redirect stderr to file
  -P, --progress                    Print dots each second while waiting
  -h, --help                        Display this help output
  -C, --nocolor                     Disable color output
  -V, --version                     Display software version
  -v, --verbose                     Run verbosely
  -q, --quiet                       Run quietly (repeat to suppress errors)     
```